SpiffWorkflow/SpiffWorkflow/bpmn/specs/events/event_types.py

```diff
@@ -36,7 +36,7 @@ def __init__(self, wf_spec, name, event_definition, **kwargs):
 
     def catches(self, my_task, event_definition, correlations=None):
         if self.event_definition == event_definition:
-            return all([ correlations.get(key) == my_task.workflow.correlations.get(key) for key in correlations ])
+            return all([ str(correlations.get(key)) == str(my_task.workflow.correlations.get(key)) for key in correlations ])
```

Comment:
Still not loving this, but don't have a resultion to it at the moment.

Reply:
For the record, I'm completely appalled at this.

Reply:
because you think the values should be exact matches even including types? you like the idea of having integer values, etc?

Reply:
Are you addressing this question to me? I don't GAF about types but I do care about not silently changing data-related behavior.

If you DO care about types, then this could cause problems.

Even you're agnostic about type checking, this means that you have to be sure objects that might return `True` when equality is tested also return `True` when string equality is tested.   If you haven't implemented  a `__str__` method on your object you're going to get `obj.__repr__`, which by default is `<module.class at address>` -- probably different even for workflows internal to SpiffWorkflow and almost guaranteed to be different otherwise.

We don't implement custom data types, so we don't care, but if I was a user, I'd use them, and I'd be irritated about this.

Reply:
yeah, i was just trying to understand why you were appalled. :) i assume the reason dan changed this was because something was returning "1" and something else was returning 1 (or similar), and by his estimation they were supposed to match, so he coerced them to strings so they'd match. i'm honestly not sure if we should expect/allow correlation values to have non-string types or not. it doesn't seem terrible to me either way. i can understand why someone might be surprised by the coercing behavior. i hadn't considered the case where coercing to strings might actually cause something to NOT match that would have otherwise matched. that would be annoying. i guess you could test both to avoid that:

```` python
(val1 == val2) or (str(val1) == str(val2))
````

Reply:
Well, that solves the case where you are agnostic about types, but not the one where you do care, because you'll get a spurious match on 1 == '1'.

Even if I don't care much about type checking (I actually do care about types, I just think type checking is overrated), I'm almost certainly in the minority on this.  Since lots of people do care, I think we should do our best not to make things harder for them.

Reply:
I think I can resolve this today.